osdk/tools/gen_dockerfile.sh

```diff
@@ -47,4 +56,34 @@ RUN cargo install cargo-binutils
 
 VOLUME [ "/root/asterinas" ]
 
-WORKDIR /root/asterinas
+WORKDIR /root/asterinas'
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --help)
+            echo "Usage: gen_dockerfile.sh [OPTIONS]"
+            echo ""
+            echo "Options:"
+            echo "  --help          Print the help information"
+            echo "  --intel-tdx     Include Intel TDX support"
+            echo "  --out-dir       Output the Dockerfile under this directory. By default, the"
+            echo "                  output directory is the current working directory."
+            exit 0
+            ;;
+        --intel-tdx)
+            dockerfile_template=$(echo "$dockerfile_template" | \
+            sed 's|FROM ubuntu:22.04|FROM intelcczoo/tdvm:ubuntu22.04-mvp_2023ww15|')
+            ;;
+        --out-dir)
+            shift
+            output_dir=$1
+            ;;
+        *)
+            echo "Unknown option: $1"
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+echo "$dockerfile_template" > "${output_dir}/Dockerfile"
```

Comment:
Using `sed` works, but is not quite intuitive. This is how to do it in a more readable way.

````bash
output_dir="."
base_image="ubuntu:22.04"

while [[ "$#" -gt 0 ]]; do
    case $1 in
        --help)
            ...
        --intel-tdx)
            base_image = "intelcczoo/tdvm:ubuntu22.04-mvp_2023ww15"
            ;;
    esac
    shift
done

docker_file = $(cat <<EOF
FROM $(base_image)

ARG DEBIAN_FRONTEND=noninteractive
RUN apt update \ 
    && apt install -y \ 
    build-essential \ 
    curl \ 
    ...

EOF
)
echo "$docker_file" > "${output_dir}/Dockerfile"
```